File: Documentation/5.4/Samples/python/ClientApi/Session/Querying/TextSearch/StartsWith.py
from examples_base import ExampleBase, Product


class StartsWith(ExampleBase):

    def setUp(self):
        super().setUp()

    def test_example(self):
        with self.embedded_server.get_document_store("StartsWith") as store:
            with store.open_session() as session:
                # region startsWith_1
                products = list(
                    session.query(object_type=Product)
                    # Call 'where_starts_with' on the field
                    # Pass the prefix to search by
                    .where_starts_with("Name", "Ch")
                )

                # Results will contain only Product documents having a 'Name' field
                # that starts with any case variation of 'ch'
                # endregion

            with store.open_session() as session:
                # region startsWith_4
                products = list(
                    session.query(object_type=Product)
                    # Pass 'exact=True' to search for an EXACT prefix match
                    .where_starts_with("Name", "Ch", exact=True)
                )

                # Results will contain only Product documents having a 'Name' field
                # that starts with 'Ch'
                # endregion

            with store.open_session() as session:
                # region startsWith_7
                products = list(
                    session.query(object_type=Product)
                    # Negate next statement
                    .not_()
                    # Call 'where_starts_with' on the field
                    # Pass the prefix to search by
                    .where_starts_with("Name", "Ch")
                )
                # Results will contain only Product documents having a 'Name' field
                # that does NOT start with 'ch' or any other case variations of it
                # endregion

File: Documentation/5.4/Samples/php/ClientApi/Session/Querying/TextSearch/StartsWith.php
<?php

use RavenDB\Samples\Infrastructure\DocumentStoreHolder;
use RavenDB\Samples\Infrastructure\Orders\Product;

class StartsWith
{
    public function examples(): void
    {
        $store = DocumentStoreHolder::getStore();
        try {
            $session = $store->openSession();
            try {
                # region startsWith_1
                /** @var array<Product> $products */
                $products = $session
                    ->query(Product::class)
                    // Call 'StartsWith' on the field
                    // Pass the prefix to search by
                    ->whereStartsWith("Name", "Ch")
                    ->toList();
                # endregion
            } finally {
                $session->close();
            }

            $session = $store->openSession();
            try {
                # region startsWith_3
                /** @var array<Product> $products */
                $products = $session->advanced()
                    ->documentQuery(Product::class)
                    // Call 'WhereStartsWith'
                    // Pass the document field and the prefix to search by
                    ->whereStartsWith("Name", "Ch")
                    ->toList();
                # endregion
            } finally {
                $session->close();
            }

            $session = $store->openSession();
            try {
                # region startsWith_4
                /** @var array<Product> $products */
                $products = $session->advanced()
                    ->query(Product::class)
                    // Pass 'exact: true' to search for an EXACT prefix match
                    ->whereStartsWith("Name", "Ch", true)
                    ->toList();
                # endregion
            } finally {
                $session->close();
            }

            $session = $store->openSession();
            try {
                # region startsWith_6
                /** @var array<Product> $products */
                $products = $session->advanced()
                    ->documentQuery(Product::class)
                    // Call 'WhereStartsWith'
                    // Pass 'exact: true' to search for an EXACT prefix match
                    ->whereStartsWith("Name", "Ch", true)
                    ->toList();
                # endregion
            } finally {
                $session->close();
            }

            $session = $store->openSession();
            try {
                # region startsWith_7
                /** @var array<Product> $products */
                $products = $session
                    ->query(Product::class)
                    # Negate next statement
                    ->not()
                    // Call 'StartsWith' on the field
                    // Pass the prefix to search by
                    ->whereStartsWith("Name", "Ch")
                    ->toList();
                # endregion
            } finally {
                $session->close();
            }

            $session = $store->openSession();
            try {
                # region startsWith_9
                /** @var array<Product> $products */
                $products = $session->advanced()
                    ->documentQuery(Product::class)
                    // Call 'Not' to negate the next predicate
                    ->not()
                    // Call 'WhereStartsWith'
                    // Pass the document field and the prefix to search by
                    ->whereStartsWith("Name", "Ch")
                    ->toList();
                # endregion
            } finally {
                $session->close();
            }
        } finally {
            $store->close();
        }
    }
}
